src/main/java/com/fundynamic/d2tm/game/math/BoundedVector2D.java

```diff
@@ -0,0 +1,33 @@
+package com.fundynamic.d2tm.game.math;
+
+
+public class BoundedVector2D<T extends Number> extends Vector2D {
```

Comment:
I would say the BoundedVector is still a vector. You can go back to a vector but it would not be 'restricted' in movement anymore. 

But that made me think. This is not about the vector at all, simply about an area you can move around in. So perhaps it would be even better to have a "Perimeter" class defining the boundaries, and then apply the vector there which gets corrected when it is over the edge.